### PR TITLE
En_Zov

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c [hunk 1]
```diff
@@ -37,63 +39,598 @@ const ActorInit En_Zov_InitVars = {
...
+        this->unk_32C = -1;
+    }
+}
+
+void func_80BD1440(EnZov* this, s16 arg1) {
+    func_80BD13DC(this);
+    this->unk_32C = arg1;
+}
+
+void func_80BD1470(EnZov* this, s16 idx, u8 mode, f32 transitionRate) {
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Zov/z_en_zov.c:149]: We prefer `index` over `idx` (applies elsewhere).

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c [hunk 2]
```diff
@@ -37,63 +39,598 @@ const ActorInit En_Zov_InitVars = {
...
+            this->unk_320 |= 2;
+            this->unk_2EE = 3;
+        }
+
+        if (this->unk_324 > 0) {
+            this->unk_324--;
+        } else {
+            func_80BD13DC(this);
+            func_801518B0(globalCtx, 0x1021, &this->actor);
+            ((Tempp*)GET_PLAYER(globalCtx)->tatlActor)->unk_264 |= 4;
```
Comment (EllipticEllipsis) [inline src/overlays/actors/ovl_En_Zov/z_en_zov.c:322]: This is an `EnElf`.

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c [hunk 3]
```diff
@@ -37,63 +40,593 @@ const ActorInit En_Zov_InitVars = {
...
+        } else {
+            Math_SmoothStepToS(&this->unk_2F0.x, 0, 6, 0x1838, 0x64);
+        }
+        Math_SmoothStepToS(&this->unk_2F0.y, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2F6.x, 0, 6, 0x1838, 0x64);
+        Math_SmoothStepToS(&this->unk_2F6.y, 0, 6, 0x1838, 0x64);
+    }
+
+    if (DECR(this->unk_2EE) == 0) {
+        this->unk_2EE = Rand_S16Offset(0x3C, 0x3C);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zov/z_en_zov.c:497]:
```suggestion
        this->unk_2EE = Rand_S16Offset(60, 60);
```
This looks like a timer

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c [hunk 4]
```diff
@@ -37,63 +40,593 @@ const ActorInit En_Zov_InitVars = {
...
+}
+
+void EnZov_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnZov* this = THIS;
+    Gfx* gfx;
+    s32 curFrame;
+    s32 phi_a1;
+    u8 phi_v1;
+
+    if (0) {}
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Zov/z_en_zov.c:558]:
```suggestion
    if (1) {}
```
this is preferred over `if (0) {}` if it still matches

**Review by engineer124 — COMMENTED** [review]
Some small comments

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c (hunk 4 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Zov/z_en_zov.c:558]: I thought 0 was preferred over 1, because it's even less intrusive?

Comment (engineer124) [reply src/overlays/actors/ovl_En_Zov/z_en_zov.c]: In what sense do you mean intrusive? My understanding was that `if (1) {}` was preferred just for consistency's sake

**Review by engineer124 — APPROVED** [review]

**Review by EllipticEllipsis — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

File: src/overlays/actors/ovl_En_Zov/z_en_zov.c (hunk 4 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Zov/z_en_zov.c]: I mean, `if (1) {}` is essentially trying to say "always include this." Whereas `if (0) {}` is saying "never include this." So if(0) seems preferable if possible. Someone's told me to do exactly this too btw, but I don't remember who. Either Anghelo or Elliptic. I don't really mind either way, but I was asked to change it before and that was the reasoning as far as I remember.

Comment (EllipticEllipsis) [reply src/overlays/actors/ovl_En_Zov/z_en_zov.c]: We usually use 1 unless there's reason to use 0.

**Pull request merged**